### ISSUE TITLE
Object deletion removes related activities

### ISSUE THREAD

**Issue by niknokseyer** [justquick/django-activity-stream#346]
Is there a way to not delete the activities even if I delete the object?

**Comment by jtrain** [comment]
To be clear why this is happening, it is because of `GenericRelation`.
https://docs.djangoproject.com/en/2.0/ref/contrib/contenttypes/#django.contrib.contenttypes.fields.GenericForeignKey

Django docs on `GenericRelation` say:

```
If the related object is deleted, the content_type and object_id
fields remain set to their original values and the GenericForeignKey returns None:

>>> guido.delete()
>>> t.content_object  # returns None
```

 `actstream` in its `registry.py` file contributes a `GenericRelation` to `actor`, `target` and `action_object`. The solution would involve not doing that. Perhaps with some option during registration? Perhaps by removing the generic relation yourself after registration.

**Comment by jtrain** [comment]
If you were very keen on straight-up deleting the generic relation, you could in the `apps.py` file where you register with `actstream`.

```
register.register(User)

def not_actor_actions(field):
  return field.name != 'actor_actions'

User._meta.private_fields = list(filter(not_actor_actions, User._meta.private_fields))
```

I tried this and it prevented the cascade delete, but I didn't put it in production, you'd want to test it for yourself

**Comment by tamhv** [comment]
> If you were very keen on straight-up deleting the generic relation, you could in the `apps.py` file where you register with `actstream`.
> 
> ```
> register.register(User)
> 
> def not_actor_actions(field):
>   return field.name != 'actor_actions'
> 
> User._meta.private_fields = list(filter(not_actor_actions, User._meta.private_fields))
> ```
> 
> I tried this and it prevented the cascade delete, but I didn't put it in production, you'd want to test it for yourself

thanks @jtrain it works, by this way i can show action for deleted `action_object`
justquick closed issue abc on django-activity-stream

after deleted the action_object...
justquick closed ~Issue #213~ on django-activity-stream
